tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+
+> There is no new Radius specific user experience for this scenario. Customers are only interacting with Flux or ArgoCD.
+
+### Detailed Customer Experience
+ <!-- <List of steps the customer goes through from the start to the end of the scenario to provide more detailed view of exactly what the customer is able to do given the new capabilities>  -->
+<!-- Step 1
+Step 2
+… -->
+
+**Step 1: Define and deploy Radius Environment and Recipes**
+
+As an operator, I need to define and deploy a Radius Environment and associate relevant Recipes to that environment so that my development team can build and deploy their Radius applications.
+
+1. Using my editor of choice, such as VSCode, I open an existing Radius environment file, `env.bicep`, from the online Radius samples repo. 
+1. I use Radius documentation to determine specific edits to my `env.bicep` file such as Kubernetes cluster and the appropriate cloud provider registrations. 
+   1. I will create a Radius Environment resource and add all the information listed above. 
+   1. I will then edit my Radius Environment resource to contain the properties required to register my Radius Recipe such as template path, template kind, etc.
+1. I then push my new `env.bicep` file to my git repository, as I would any other file. 
+1. This triggers Flux or ArgoCD to detect the change, read the new `env.bicep` file, and deploy the Radius Environment as specified.
+1. After Flux or ArgoCD initiates the Radius environment deployment, as I would normally do, I use Flux or ArgoCD to monitor deployment status and health status of the underlying Kubernetes cluster. 
+
+> As an operator, while Radius presents some new concepts, those concepts fit cleanly within my existing GitOps workflows so Radius is relatively low overhead for me to adopt and to provide value to my development counterparts.   
+
+**Step 2: Deploy and manage Radius Applications and Resources through GitOps toolset**
+
+Now, as a developer, I need to make some changes to my application code. Specifically, I need to change the Radius Resource for my frontend UI. My frontend resource is already running in my Radius application but I need update the code for parsing user input. First, I'll update the required file where this function lives then I'll update my Radius Resource as follows: 
+
+1. With my editor of choice, such as VSCode, I open the existing `frontend` application file `helper.ts` and update my function `parseUsers` with the new required logic. 
```

Comment:
I think maybe we should move this one to a different scenario and talk here about how app developers could edit their `app.bicep` Radius application and the changes would be applied to the cluster. in this scenario the users would have to re-build their container image for their application and we would have to re-apply the Radius app deployment or something. let's discuss in the meeting

Reply:
Clarify the scope of gitops: it doesn't manage the CI (e.g. image build and app containerization) but it does manage the CD (e.g. deployment of the app to the cluster)